source/common/secret/sds_api.h

```diff
@@ -43,14 +45,11 @@ class SdsApi : public Init::Target,
     return MessageUtil::anyConvert<envoy::api::v2::auth::Secret>(resource).name();
   }
 
-  // SecretProvider
-  const Ssl::TlsCertificateConfig* secret() const override {
-    return tls_certificate_secrets_.get();
-  }
-
-  Common::CallbackHandle* addUpdateCallback(std::function<void()> callback) override {
-    return update_callback_manager_.add(callback);
-  }
+protected:
+  // Creates new secrets.
+  virtual void set_secret(const envoy::api::v2::auth::Secret&) PURE;
+  Common::CallbackManager<> update_callback_manager_;
+  uint64_t secret_hash_;
```

Comment:
Why is `secret_hash_` no longer private?

Reply:
It should be private, thanks for catching this.
It was changed to be protected because it was assigned to a new hash in derived class. But then we simplified that part of code and moved hash calculation back to base class.